### ISSUE TITLE
HTTP API v2 - more RESTful handling of data

### ISSUE THREAD

**Issue by pferreir** [indico/indico#1773]
In a next version, we should be able to access API information is a more RESTful way.

(Purely arbitrary) example
 * `/category/0` - returns tree of sub-categories
 * `/category/2` - returns events
 * `/category/0/events` - returns joined events of sub-categories

Other things to improve:
  * API versioning (backwards-compatible)
  * Limits and offsets (can use relational!)
  * `Accept` headers instead of extensions (like everyone else)
  * Using `/api/v2` instead of `/export` (redirect `/export` -> `/api/v1` for now)

**Comment by pferreir** [comment]
See also #1772

**Comment by ThiefMaster** [comment]
Accept headers are useful, but then we'd need separate endpoints to get e.g. ical links for people to paste in calendar apps. Wouldn't be a big deal though..

**Comment by pferreir** [comment]
Yeah. I wouldn't be surprised if iCal, etc... do not send proper Accept headers. But we can easily proxy the API using a custom handler.

**Comment by pferreir** [comment]
Also worth considering: using UTC timestamp instead of local server time.

**Comment by gordonwatts** [comment]
Just to make explicit something I sent around so it is attached here: it would be good to version the JSON that is returned. For example, if your database internal structure changes, the JSON that is returned will change format. Code I write currently uses heuristics to figure out what is the proper parsing strategy (see https://github.com/gordonwatts/IndicoInterface.NET for a C# version of this code that tries to work across various versions of indico... and has, unfortunately, grown organically). Having some sort of version number in the returned JSON would be a big help.

The use case is as follows: there are a lot of indico servers out there, and they are in various stages of upgrades matching (or not) where CERN is. The result is if you want some code that can extract meeting information from more than just CERN, you have to have some way of figuring out what the current format is (XML, JSON, which version of JSON (though I think there is currently only one, but there will be multiple ones as the database is further normalized).

**Comment by mvidalgarcia** [comment]
New API should allow creation/deletion/moving/cloning of events.

`RQF1167107`

**Comment by ThiefMaster** [comment]
I'd drop anything but JSON when it comes to formats. Pretty much all APIs nowadays just support JSON and if someone really needs XML they can always deploy a microservice to convert it.

Related: #2345 (avoid using the API for private URLs generated by indico itself)
